wgpu-hal/src/metal/adapter.rs

```diff
@@ -902,6 +902,18 @@ impl super::PrivateCapabilities {
                 && (device.supports_family(MTLGPUFamily::Apple7)
                     || device.supports_family(MTLGPUFamily::Mac2)),
             supports_shared_event: version.at_least((10, 14), (12, 0), os_is_mac),
+            supported_vertex_amplification_factor: {
+                let mut factor = 1;
+                // https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf#page=8
+                // The table specifies either none, 2, 8, or unsupported, implying it is a relatively small power of 2
+                // The bitmask only uses 32 bits, so it can't be higher even if the device for some reason claims to support that.
+                loop {
+                    if factor >= 32 || !device.supports_vertex_amplification_count(factor * 2) {
+                        break factor as u32;
+                    }
+                    factor *= 2
```

Comment:
why not a while loop?
```suggestion
                while factor < 32 && device.supports_vertex_amplification_count(factor * 2) {
                    factor *= 2
```